server/komodo-rest/src/main/java/org/komodo/rest/service/KomodoConnectionService.java

```diff
@@ -377,6 +387,129 @@ public Response createConnection( final @Context HttpHeaders headers,
+                                        @ApiParam(
+                                              value = "" + 
+                                                      "Properties for the new connection:<br>" +
+                                                      OPEN_PRE_TAG +
+                                                      OPEN_BRACE + BR +
+                                                      NBSP + "description: \"description for the connection\"" + COMMA + BR +
+                                                      NBSP + "serviceCatalogSource: \"serviceCatalog source for the connection\"" + BR +
+                                                      NBSP + CLOSE_BRACE + BR +
+                                                      CLOSE_BRACE +
+                                                      CLOSE_PRE_TAG,
+                                              required = true
+                                        )
+                                        final String connectionJson) throws KomodoRestException {
+
+        SecurityPrincipal principal = checkSecurityContext(headers);
+        if (principal.hasErrorResponse())
+            return principal.getErrorResponse();
+
+        List<MediaType> mediaTypes = headers.getAcceptableMediaTypes();
+        if (! isAcceptable(mediaTypes, MediaType.APPLICATION_JSON_TYPE))
+            return notAcceptableMediaTypesBuilder().build();
+
+        // Error if the connection name is missing
+        if (StringUtils.isBlank( connectionName )) {
+            return createErrorResponseWithForbidden(mediaTypes, RelationalMessages.Error.CONNECTION_SERVICE_CREATE_MISSING_NAME);
+        }
+
+        ServiceCatalogDataSource serviceCatalogSource = null;
+        
+        RestConnection restConnection = new RestConnection();
+        restConnection.setId(connectionName);
+
+        final KomodoConnectionServiceConnAttributes rcAttr;
+        try {
+            rcAttr = KomodoJsonMarshaller.unmarshall(connectionJson, KomodoConnectionServiceConnAttributes.class);
+
+            // Add properties for the description and serviceCatalogSource
+            restConnection.addProperty("description", rcAttr.getDescription());
+            restConnection.addProperty("serviceCatalogSource", rcAttr.getServiceCatalogSource());
+            restConnection.setJdbc(true);
+            
+            // Get the specified ServiceCatalogDataSource from the metadata instance
+            Collection<ServiceCatalogDataSource> dataSources = getMetadataInstance().getServiceCatalogSources();
+			for(ServiceCatalogDataSource ds: dataSources) {
+				if(ds.getName().equals(rcAttr.getServiceCatalogSource())) {
+					serviceCatalogSource = ds;
+					break;
+				}
+			}
+			// If catalogSource is not found, exit with error
+			if(serviceCatalogSource == null) {
+                return createErrorResponseWithForbidden(mediaTypes, RelationalMessages.Error.CONNECTION_SERVICE_CATALOG_SOURCE_DNE_ERROR);
```

Comment:
Really really minor point -> can this method be space/tab formatted?

Reply:
yes